src/online2bin/online2-tcp-nnet3-decode-faster.cc

```diff
@@ -0,0 +1,462 @@
+      am_nnet.Read(ki.Stream(), binary);
+      SetBatchnormTestMode(true, &(am_nnet.GetNnet()));
+      SetDropoutTestMode(true, &(am_nnet.GetNnet()));
+      nnet3::CollapseModel(nnet3::CollapseModelConfig(), &(am_nnet.GetNnet()));
+    }
+
+    // this object contains precomputed stuff that is used by all decodable
+    // objects.  It takes a pointer to am_nnet because if it has iVectors it has
+    // to modify the nnet to accept iVectors at intervals.
+    nnet3::DecodableNnetSimpleLoopedInfo decodable_info(decodable_opts,
+                                                        &am_nnet);
+
+    KALDI_LOG << "Loading FST...";
+
+    fst::Fst<fst::StdArc> *decode_fst = ReadFstKaldiGeneric(fst_rxfilename);
+
+    fst::SymbolTable *word_syms = NULL;
+    if (!word_syms_rxfilename.empty())
+      if (!(word_syms = fst::SymbolTable::ReadText(word_syms_rxfilename)))
+        KALDI_ERR << "Could not read symbol table from file "
+                  << word_syms_rxfilename;
+
+    KALDI_LOG << "Loaded evertyhing. Waiting for data...";
+
+    signal(SIGPIPE, SIG_IGN); //ignore SIGPIPE to avoid crashing when socket forecfully disconnected
+
+    TcpServer server(read_timeout);
+
+    server.Listen(port_num);
+
+    while (true) {
+
+      server.Accept();
+
+      int32 samp_count = 0;//this is used for output refresh rate
+      size_t chunk_len = static_cast<size_t>(chunk_length_secs * samp_freq);
+
+      int32 check_update = static_cast<int32>(samp_freq * output_freq);
+      int32 next_check = check_update;
+
+      int32 frame_offset = 0;
+
+      bool eos = false;
+
+      OnlineIvectorExtractorAdaptationState adaptation_state(
+          feature_info.ivector_extractor_info);
+
+      OnlineNnet2FeaturePipeline feature_pipeline(feature_info);
+      SingleUtteranceNnet3Decoder decoder(decoder_opts, trans_model,
+                                          decodable_info,
+                                          *decode_fst, &feature_pipeline);
+
+      while (!eos) {
+
+        decoder.InitDecoding(frame_offset);
+
+//        if (!adapt_speaker)
+//          feature_pipeline.SetAdaptationState(adaptation_state); //reset adaptation state to intital one
+
+        OnlineSilenceWeighting silence_weighting(
+            trans_model,
+            feature_info.silence_weighting_config,
+            decodable_opts.frame_subsampling_factor);
+
+        std::vector<std::pair<int32, BaseFloat>> delta_weights;
+
+        int32 samp_pipeline = 0;//this is used to figure out the offset for the remainder
+
+        while (true) {
+
+          eos = !server.ReadChunk(chunk_len);
+
+          if (!eos) {
+            Vector<BaseFloat> wave_part = server.GetChunk();
+            feature_pipeline.AcceptWaveform(samp_freq, wave_part);
+            samp_pipeline += wave_part.Dim();
+            samp_count += chunk_len;
+
+            if (silence_weighting.Active() &&
+                feature_pipeline.IvectorFeature() != NULL) {
```

Comment:
We still need the first two lines that used be here. I mean the lines which compute delta weights.

Reply:
Yea, my bad. I don't really use silence weighing in my model so I missed that in my tests :(

Reply:
Did you have an opportunity to check if silence weighting works correctly?